src/ILCompiler.ReadyToRun/src/Compiler/ReadyToRunMetadataFieldLayoutAlgorithm.cs

```diff
@@ -741,12 +743,12 @@ public static bool IsManagedSequentialType(TypeDesc type)
             {
                 foreach (FieldDesc field in type.GetFields())
                 {
-                    if (!field.IsStatic && !field.IsLiteral)
+                    if (!field.IsStatic &&
+                        !field.IsLiteral &&
+                        !field.FieldType.IsEnum &&
```

Comment:
Enum handling feels a bit odd. Is this really how CoreCLR handles this?

An enum type is not considered managed sequential, but if a struct has an enum-typed field, it _is_ managed sequential?

(If yes, maybe add a comment.)

Reply:
That is correct. Enums have Auto-layout as per ECMA-335. When calculating if a struct is ManagedSequential, CoreCLR normalizes each field in the field signature, which (among other things) transforms enum types to their underlying value. So structs containing an enum-typed field are considered managed-sequential.

Reply:
Based on Jeremy's feedback I have modified the formulation so that the recursion over fields performs the TypeIsManagedSequential check for the underlying field type, that seems to model CoreCLR runtime behavior relatively closely.